docs: Add hybrid search documentation (#457)
googleapis/langchain-google-alloydb-pg-python: README.rst

       embedding_service=embeddings_service
   )

> Hybrid Search with AlloyDBVectorStore

Reviewer: ```suggestion
Hybrid Search with AlloyDBVectorStore
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

With AlloyDBVectorStore you can use hybrid search for more comprehensive and relevant search results.

> .. code-block:: python
> 
>   vs = AlloyDBVectorStore.create_sync(
>       engine=engine,

Reviewer: ```suggestion
.. code-block:: python
  from langchain_google_alloydb_pg import HybridSearchConfig, reciprocal_rank_fusion
  
  vs = AlloyDBVectorStore.create_sync(
      engine=engine,
```

       embedding_service=embeddings_service
   )

> Hybrid Search with AlloyDBVectorStore
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
Hybrid Search
^^^^^^^^^^^
```

> With AlloyDBVectorStore you can use hybrid search for more comprehensive and relevant search results.

Reviewer: ```suggestion
The `AlloyDBVectorStore` supports hybrid search (dense vectors + full text) for more comprehensive and relevant search results.
```
